pupil-spa/src/app/connectivity-check/connectivity-check.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { ConnectivityService } from '../services/connectivity-service/connectivity-service';
+import { APP_CONFIG } from '../services/config/config.service';
+import { CheckStatusService } from '../services/check-status/check-status.service';
+
+@Component({
+  selector: 'app-connectivity-check',
+  templateUrl: './connectivity-check.component.html',
+  styleUrls: ['./connectivity-check.component.scss']
+})
+export class ConnectivityCheckComponent implements OnInit {
+
+  connectivityCheckViewMinDisplay;
+
+  constructor(
+    private connectivityService: ConnectivityService,
+    private router: Router,
+    private checkStatusService: CheckStatusService
+  ) {
+    const { connectivityCheckViewMinDisplay } = APP_CONFIG;
+    this.connectivityCheckViewMinDisplay = connectivityCheckViewMinDisplay;
+  }
+
+  /**
+   * Sleep function (milliseconds) to provide minimal display time for submission pending screen
+   * @param {Number} ms
+   * @returns {Promise.<void>}
+   */
+  private sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
+
+  async ngOnInit() {
+    const hasUnfinishedCheck = this.checkStatusService.hasUnfinishedCheck();
+    if (hasUnfinishedCheck) {
+      return this.router.navigate(['check'], { queryParams: { unfinishedCheck: true } });
+    }
+    const startTime = Date.now();
+    const connectivityCheckSucceeded = await this.connectivityService.connectivityCheckSucceeded();
```

Comment:
I think that as `connectivityCheckSucceeded` may throw an error (when rejecting a Promise, if for instance the check fails) we should try/catch it here to prevent `ngOnInit` also throwing an error/rejecting as it is also `async`

Reply:
resolved based on conversation